iroha/src/lib.rs

```diff
@@ -296,6 +304,26 @@ impl Iroha {
         );
         Ok(())
     }
+
+    /// To wait for active peers(this is provisional code)
+    async fn wait_for_active_peers(&self, sorted_peers: &[PeerId]) -> Result<(), String> {
+        log::info!("Waiting for active peers");
+        for peer in sorted_peers.iter() {
+            let url = httpUri::Builder::new()
+                .scheme("http")
+                .authority(peer.address.as_str())
+                .path_and_query(uri::HEALTH_URI)
+                .build()
+                .expect(&format!("invalid peer address {}", peer.address.as_str()));
+            let response = isahc::get(url.to_string())
```

Comment:
Please use `isahc::get_async` to use asynchronous Rust syntax https://docs.rs/isahc/1.0.3/isahc/#asynchronous-requests

Reply:
You also don't need to convert url to string. Just pass url itself and isahc will accept it like that: `isahc::get_async(url).await`